crates/core/src/network/connection_manager.rs

```diff
@@ -592,6 +592,7 @@ where
                             connection.downgrade(),
                             header.msg_id,
                             header.in_response_to,
+                            PeerMetadataVersion::from(&header),
```

Comment:
Why not moving it? `from` consumes its input by convention.

Reply:
I was thinking in case header is still needed in some other code afterwards. Since all version(s) are copy anyway.